### PR TITLE
[Security] Document the new remember me system

### LINKED ISSUE #15721: [Security] Deprecate legacy remember me services
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#42578
| PR author(s) | @wouterj
| Merged in    | 5.4

### LINKED ISSUE #16149: RememberMeBadge in remember Me Functionality
Hello,

I followed the documentation to set up the remember me functionality and the remember me cookie is not created during authentication if in addition I do not add 'new RememberMeBadge ()' in the authenticate function of the AppCustomAuthenticator .

[](https://symfony.com/doc/current/security/remember_me.html#forcing-the-user-to-re-authenticate-before-accessing-certain-resources)

```php
public function authenticate(Request $request): PassportInterface
{
    $email = $request->request->get('email', '');

    $request->getSession()->set(Security::LAST_USERNAME, $email);

    return new Passport(
        new UserBadge($email),
        new PasswordCredentials($request->request->get('password', '')),
        [
            new CsrfTokenBadge('authenticate', $request->request->get('_csrf_token')),
            new RememberMeBadge()
        ]
    );
}
```
thank you for your feedback and your actions

### PR DESCRIPTION
**Pull request by wouterj**
Fixes #15721, fixes #16149
Replaces #15464, replaces #15893
Ref #15908

In Symfony 5.3, along with the new authentication system, we also somewhat silently introduced a new remember me system. Time to update the remember me docs with all new features :)

This PR also includes 2 commits from open PRs in this repository. We've been very late with this one, and community members have already invested time in contributing necessary changes to the docs before. They deserve the credits :)

@javiereguiluz I'm sorry for not informing you of this work before, I see you just merged a third PR that is replaced by this one.

### CODE REVIEW

File: security/remember_me.rst [hunk 1]
```diff
@@ -81,223 +86,404 @@ the session lasts using a cookie with the ``remember_me`` firewall option:
...
 
-    There is also a ``IS_REMEMBERED`` attribute that grants access *only* when the
-    user is authenticated via the remember me mechanism.
+Symfony provides two ways to validate remember me tokens:
 
-.. versionadded:: 5.1
+Signature based tokens
+    By default, the remember me cookie contains a signature based on
+    properties of the user. If the properties change, the signature changes
+    and the tokens is no longer considered valid. See
```
Comment (gnito-org) [inline security/remember_me.rst]: Should be:

``and the tokens are no longer considered valid.``

File: security/remember_me.rst [hunk 2]
```diff
@@ -81,223 +86,404 @@ the session lasts using a cookie with the ``remember_me`` firewall option:
...
+Using Signed Remember Me Tokens
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, remember me cookies contain a *hash* that is used to validate
+the cookie. This hash is computed based on configured
+signature properties.
+
+These properties are always included in the hash:
+
+* The user idenifier (returned by
```
Comment (gnito-org) [inline security/remember_me.rst]: Should be:

``The user identifier...``

File: security/remember_me.rst [hunk 3]
```diff
@@ -81,223 +86,404 @@ the session lasts using a cookie with the ``remember_me`` firewall option:
...
+                    ->secret('%kernel.secret%')
+                    // ...
+                    ->signatureProperties(['password', 'updatedAt'])
+            ;
+        };
 
-Then you need to create a table with the following structure in your database
-so ``DoctrineTokenProvider`` can store the tokens:
+In this example, the remember me cookie will no longer be considered valid
+if the ``updatedAt``, password or user identifier for this user change.
```
Comment (gnito-org) [inline security/remember_me.rst]: Should be:

``...for this user changes.``

File: security/remember_me.rst [hunk 4]
```diff
@@ -81,223 +86,404 @@ the session lasts using a cookie with the ``remember_me`` firewall option:
...
+.. _security-remember-me-storage:
 
-        // ...
-    }
+Customize how Remember Me Tokens are Stored
+-------------------------------------------
 
-.. tip::
+Remember me cookies contain a token that is used to verify the user's
+identity. As these tokens are longlived, it is important to take
```
Comment (gnito-org) [inline security/remember_me.rst]: Should be:

``long-lived``

File: security/remember_me.rst [hunk 5]
```diff
@@ -81,223 +86,404 @@ the session lasts using a cookie with the ``remember_me`` firewall option:
...
 
-.. note::
+.. _security-remember-me-persistent:
+
+Storing Remember Me Tokens in the Database
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    If you use DoctrineMigrationsBundle to manage your database migrations, you
-    will need to tell Doctrine to ignore this new ``rememberme_token`` table:
+As remember me tokens are often longlived, you might prefer to save them in
```
Comment (gnito-org) [inline security/remember_me.rst]: Should be:

``long-lived``

**Comment by wouterj** [comment]
Thanks for the detailed review @gnito-org

File: security/remember_me.rst [hunk 6]
```diff
@@ -81,223 +86,404 @@ the session lasts using a cookie with the ``remember_me`` firewall option:
...
+--------------------------------------------
+
+Not all authentication methods support remember me (e.g. HTTP Basic
+authentication doesn't have support). An authenticator indicates support
+using a ``RememberMeBadge`` on the :ref:`security passport <security-passport>`.
+
+After logging in, you can use the security profiler to see if this badge is
+present:
+
+.. image:: /_images/security/profiler-badges.png
```
Comment (MarkPedron) [inline security/remember_me.rst:222]: Can you add an `alt` attribute to improve accessibility?

Comment (MarkPedron) [reply security/remember_me.rst:222]:
```suggestion
.. image:: /_images/security/profiler-badges.png
   :alt: A RememberMeBadge dumped in the security profiler passport column
```

**Review by javiereguiluz — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Wouter, thanks a lot for your continuous work on Symfony Docs, especially in security-related docs 🙏 

If you agree, I'll let you merge this one because you'll deal with *upmerge* conflicts better than I'll do. Thanks!

**Comment by wouterj** [comment]
Thanks for the review, @javiereguiluz !

**Pull request merged**